File: Phase - 7 Automation Testing Using Java/Assessment-2/automationExcersice/src/test/java/com/pages/LoginPage.java
package com.pages;

import java.time.Duration;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.PageFactory;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

public class LoginPage extends BasePage {

	WebDriverWait wait;
	
    public LoginPage(WebDriver driver) {
        super(driver);
        wait = new WebDriverWait(driver, Duration.ofSeconds(10));
        PageFactory.initElements(driver, this);
    }

    @FindBy(xpath = "//input[@data-qa='signup-name']")
    private WebElement username;

    @FindBy(xpath = "//input[@data-qa='signup-email']")
    private WebElement email;

    @FindBy(xpath = "//input[@data-qa='login-email']")
    private WebElement loginEmail;

    @FindBy(xpath = "//input[@data-qa='login-password']")
    private WebElement password;

    @FindBy(xpath = "//button[@data-qa='login-button']")
    private WebElement loginButton;

    @FindBy(xpath = "//button[@data-qa='signup-button']")
    private WebElement signUpButton;
    
    private By greet = By.xpath("//a[contains(text(), \" Logged in as \")]");
    private By errorMsg = By.xpath("//p[text() = \"Your email or password is incorrect!\"]");

  

    public void setUsername(String name) {
        username.sendKeys(name);
    }

    public void setEmail(String mail) {
        email.sendKeys(mail);
    }

    public void setLoginEmail(String mail) {
        loginEmail.sendKeys(mail);
    }

    public void setPassword(String pwd) {
        password.sendKeys(pwd);
    }

    public void clickLogin() {
        loginButton.click();
    }

    public void clickSignUp() {
        signUpButton.click();
    }
    
    public String getMessage()
    {
    	return wait.until(ExpectedConditions.elementToBeClickable(greet)).getText();
    }
    
    public String getErrorMessage()
    {
    	return wait.until(ExpectedConditions.elementToBeClickable(errorMsg)).getText();
    }

    public void signUp(String name, String email)
    {
    	setUsername(name);
    	setEmail(email);
    	clickSignUp();
    }
    
    public void logIn(String name, String pass)
    {
    	setLoginEmail(name);
    	setPassword(pass);
    	clickLogin();
    }
}

File: Phase - 12 Playwright Automation/Playwright_Practice-master/Playwright_Practice-master/Cucumber_Playwright_BDD_Framework/src/test/pages/LoginPage.ts
import {Page,Locator} from '@playwright/test'
import { BasePage } from './BasePage';
export class LoginPage extends BasePage{
    readonly page:Page;
    readonly username:Locator;
    readonly password:Locator;
    readonly loginButton:Locator;
    readonly errorMessage:Locator;
    constructor(page:Page){
        super();
        this.page=page;
        this.username=page.locator("//input[@id='input-email']");
        this.password=page.locator("//input[@id='input-password']");
        this.loginButton=page.locator("//input[@value='Login']");
        this.errorMessage=page.locator("//div[@class='alert alert-danger alert-dismissible']")
    }
    async setUsername(username:string){
        await this.fill(this.username,username)
        
    }
    async setPassword(password:string){
        await this.fill(this.password,password)
    }
    async clickLogin(){
        await this.click(this.loginButton)
    }
}